src/components/common/ui/button-mui/Button.styles.ts

```diff
@@ -0,0 +1,197 @@
+import { SxProps, Theme } from '@mui/material/styles';
+export const button = (
+  color,
+  variant,
+  size,
+  startIcon,
+  endIcon,
+): SxProps<Theme> => ({
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'center',
+  borderRadius: '4px',
+  color: 'gray.600',
+  textTransform: 'none',
+  fontSize: '14px',
+  fontWeight: '700',
+  minWidth: 'unset',
+  minHeight: 'unset',
+  ...((startIcon || endIcon) && {
+    gap: '16px',
+  }),
+  ...(size == 'small' && {
+    padding: '6px 12px',
+    fontSize: '8px',
+  }),
+  ...(size == 'medium' && {
+    padding: '12px 24px',
+    fontSize: '12px',
+  }),
+  ...(size == 'large' && {
+    padding: '16px 32px',
+    fontSize: '18px',
+  }),
+  ...(variant == 'filled' &&
+    color == 'primary' && {
+      backgroundColor: 'primary.400',
+      border: `2px transparent solid`,
+      '&:hover': {
+        backgroundColor: 'primary.200',
+        border: `2px transparent solid`,
+      },
+      '&:active': {
+        backgroundColor: 'primary.logo',
+        border: `2px transparent solid`,
+      },
+      '&:focus:not(:active)': {
+        backgroundColor: 'primary.logo',
+        border: `2px solid`,
+        borderColor: 'primary.25',
+      },
+      '&:disabled': {
+        color: 'gray.300',
+        backgroundColor: 'gray.200',
+        cursor: 'none',
+        border: 'transparent',
+      },
+    }),
+  ...(variant == 'outline' &&
+    color == 'primary' && {
+      backgroundColor: 'unset',
+      border: `2px solid`,
+      borderColor: 'primary.200',
+      '&:hover': {
+        backgroundColor: 'primary.0',
+        borderColor: 'primary.200',
+      },
+      '&:active': {
+        backgroundColor: 'primary.25',
+        borderColor: 'primary.200',
+      },
+      '&:focus:not(:active)': {
+        backgroundColor: 'primary.10',
+        border: `2px solid`,
+        borderColor: 'primary.25',
+      },
+      '&:disabled': {
+        color: 'gray.200',
+        backgroundColor: 'backgroundDark.300',
+        cursor: 'none',
+        borderColor: 'gray.200',
+      },
+    }),
+  ...(variant == 'filled' &&
+    color == 'secondary' && {
+      backgroundColor: 'backgroundDark.10',
+      border: `2px transparent solid`,
+      '&:hover': {
+        backgroundColor: 'backgroundDark.50',
+        border: `2px transparent solid`,
+      },
+      '&:active': {
+        backgroundColor: 'backgroundDark.100',
+        border: `2px transparent solid`,
+      },
+      '&:focus:not(:active)': {
+        backgroundColor: 'backgroundDark.10',
+        border: `2px solid`,
+        borderColor: 'backgroundDark.100',
+      },
+      '&:disabled': {
+        color: 'gray.50',
+        backgroundColor: 'unset',
+        cursor: 'none',
```

Comment:
i think we should use `not-allowed` for disabled elements. it is more intuitive behavior